includes/REST_API/Stories_Media_Controller.php

```diff
@@ -69,6 +69,67 @@ public function get_items( $request ) {
 		return $response;
 	}
 
+	/**
+	 * Updates a single attachment.
+	 *
+	 * Override the existing method so we can set parent id.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		// WP_REST_Attachments_Controller doesn't allow setting an attachment as the parent post.
+		// Hence we are working around this here.
+		$parent_post = ! empty( $request['post'] ) ? (int) $request['post'] : null;
+		unset( $request['post'] );
+
+		if ( ! $parent_post ) {
+			return parent::update_item( $request );
+
+		}
+
+		if ( 'revision' === get_post_type( $parent_post ) ) {
+			return new WP_Error(
+				'rest_invalid_param',
+				__( 'Invalid parent type.', 'web-stories' ),
+				[ 'status' => 400 ]
+			);
+		}
+		$post_id           = $request['id'];
+		$attachment_before = $this->get_post( $post_id );
+		if ( is_wp_error( $attachment_before ) ) {
+			return $attachment_before;
+		}
+		$args   = [
+			'ID'          => $post_id,
+			'post_parent' => $parent_post,
+		];
+		$result = wp_update_post( $args, true );
+
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		$attachment = $this->get_post( $post_id );
+		if ( is_wp_error( $attachment ) ) {
+			return $attachment;
+		}
+
+		$request->set_param( 'context', 'edit' );
+
+		/** This action is documented in wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php */
+		do_action( 'rest_after_insert_attachment', $attachment, $request, false );
+
+		/** This action is documented in wp-includes/post.php */
+		do_action( 'wp_after_insert_post', $post_id, $attachment, true, $attachment_before );
+
+		$response = $this->prepare_item_for_response( $attachment, $request );
+
+		return rest_ensure_response( $response );
```

Comment:
I was hoping we could get away with less duplication.

Could you help me understand why this was necessary here?

Reply:
I am not sure what you mean. We want to maintain the actions here right?

Reply:
Yes, but when I wrote https://github.com/google/web-stories-wp/pull/5573#discussion_r536667098 I was hoping we could do this without duplicating the code from `WP_REST_Attachments_Controller`.

Basically:

```php
// Run parent class code
$response = parent::update_item( $request )

// Manually set parent
wp_update_post( ... )

// Ensure that $response now contains the correct post parent information
$response['post'] = $parent_id;

return $response
```

Reply:
Remove some of the repeated logic. https://github.com/google/web-stories-wp/pull/5573/commits/70d61d7cfb6fb9ee315699ec1de1535acbe89dba

The code snippet above would not work as the filter and actions would not have the correct value for the attachment object.